### PR TITLE
Sass 429

### PR DESCRIPTION
**Pull request by dnyuar**

### CODE REVIEW

File: app/controllers/predicates/TaxYearFilter.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    val config = appConfig.defaultTaxYear
+
+    if (!appConfig.taxYearErrorFeature) {
+
+      f
+
+    } else {
+
+      if (taxYear == config) {
```
Comment (BenCosfordHMRC) [inline app/controllers/predicates/TaxYearFilter.scala:40]: can be simplified to `if (taxYear == config || !appConfig.taxYearErrorFeature) {`

File: app/controllers/predicates/TaxYearFilter.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
```
Comment (BenCosfordHMRC) [inline app/controllers/predicates/TaxYearFilter.scala:58]: same as above

File: app/config/AppConfig.scala [hunk 3]
```diff
@@ -77,4 +77,6 @@ class AppConfig @Inject()(servicesConfig: ServicesConfig) {
 
   lazy val timeoutDialogTimeout: Int = servicesConfig.getInt("timeoutDialogTimeout")
   lazy val timeoutDialogCountdown: Int = servicesConfig.getInt("timeoutDialogCountdown")
+
+  lazy val taxYearErrorFeature: Boolean = false
```
Comment (BenCosfordHMRC) [inline app/config/AppConfig.scala:81]: move boolean to application.conf so we can change it via app-config-qa etc. 

Be something like servicesConfig.getBoolean("blahblah")

**Review by BenCosfordHMRC — CHANGES_REQUESTED** [review]

File: app/config/AppConfig.scala (hunk 3 above)
Comment (dnyuar) [reply app/config/AppConfig.scala]: Feature switch added to application.conf

File: app/controllers/predicates/TaxYearFilter.scala (hunk 1 above)
Comment (dnyuar) [reply app/controllers/predicates/TaxYearFilter.scala]: refactored if statement

**Review by BenCosfordHMRC — APPROVED** [review]

**Pull request merged**